File: 647-palindromic-substrings/647-palindromic-substrings.cpp
class Solution {
    int count;
    void PalCount(string s, int i, int j)
    {
        while ( i >= 0 && j < s.length() && s[i]==s[j])
        {
            --i;
            ++j;
            ++count;
        }
    }
public:
    int countSubstrings(string s) {
        int n = s.length();
        // initializing count as zero
        count = 0;
        for( int i = 0; i < n; i++ )
        {
            PalCount(s,i,i);
            cout<<i<<" "<<i<<" "<<count<<endl;
            PalCount(s,i,i+1);
            cout<<i<<" "<<i+1<<" "<<count<<endl;
        }
        return count; 
    }
};

File: 647-palindromic-substrings/647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        int count = 0;
        if(s == null || s.isEmpty()) {
            return count;
        }
        for (int i =0;i<s.length();i++) {
            for(int j = i;j<s.length();j++) {
                String subStr = s.substring(i,j+1);
                StringBuilder str1 = new StringBuilder(subStr).reverse();
                if(str1.toString().equals(subStr)) {
                    count++;
                }
            }
        }
        
        return count;
    }
}